src/_P078_Eastron.ino

```diff
@@ -285,101 +336,498 @@ float p078_readVal(byte query, byte node, unsigned int model) {
+      case 29: return SDM_FREQUENCY;
+      case 30: return SDM_IMPORT_ACTIVE_ENERGY;
+      case 31: return SDM_EXPORT_ACTIVE_ENERGY;
+      case 32: return SDM_IMPORT_REACTIVE_ENERGY;
+      case 33: return SDM_EXPORT_REACTIVE_ENERGY;
+      case 34: return SDM_VAH_SINCE_LAST_RESET;
+      case 35: return SDM_AH_SINCE_LAST_RESET;
+      case 36: return SDM_TOTAL_SYSTEM_POWER_DEMAND;
+      case 37: return SDM_MAXIMUM_TOTAL_SYSTEM_POWER_DEMAND;
+      case 38: return SDM_TOTAL_SYSTEM_VA_DEMAND;
+      case 39: return SDM_MAXIMUM_TOTAL_SYSTEM_VA_DEMAND;
+      case 40: return SDM_NEUTRAL_CURRENT_DEMAND;
+      case 41: return SDM_MAXIMUM_NEUTRAL_CURRENT;
+      case 42: return SDM_LINE_1_TO_LINE_2_VOLTS;
+      case 43: return SDM_LINE_2_TO_LINE_3_VOLTS;
+      case 44: return SDM_LINE_3_TO_LINE_1_VOLTS;
+      case 45: return SDM_AVERAGE_LINE_TO_LINE_VOLTS;
+      case 46: return SDM_NEUTRAL_CURRENT;
+      case 47: return SDM_PHASE_1_LN_VOLTS_THD;
+      case 48: return SDM_PHASE_2_LN_VOLTS_THD;
+      case 49: return SDM_PHASE_3_LN_VOLTS_THD;
+      case 50: return SDM_PHASE_1_CURRENT_THD;
+      case 51: return SDM_PHASE_2_CURRENT_THD;
+      case 52: return SDM_PHASE_3_CURRENT_THD;
+      case 53: return SDM_AVERAGE_LINE_TO_NEUTRAL_VOLTS_THD;
+      case 54: return SDM_AVERAGE_LINE_CURRENT_THD;
+      case 55: return SDM_TOTAL_SYSTEM_POWER_FACTOR_INV;
+      case 56: return SDM_PHASE_1_CURRENT_DEMAND;
+      case 57: return SDM_PHASE_2_CURRENT_DEMAND;
+      case 58: return SDM_PHASE_3_CURRENT_DEMAND;
+      case 59: return SDM_MAXIMUM_PHASE_1_CURRENT_DEMAND;
+      case 60: return SDM_MAXIMUM_PHASE_2_CURRENT_DEMAND;
+      case 61: return SDM_MAXIMUM_PHASE_3_CURRENT_DEMAND;
+      case 62: return SDM_LINE_1_TO_LINE_2_VOLTS_THD;
+      case 63: return SDM_LINE_2_TO_LINE_3_VOLTS_THD;
+      case 64: return SDM_LINE_3_TO_LINE_1_VOLTS_THD;
+      case 65: return SDM_AVERAGE_LINE_TO_LINE_VOLTS_THD;
+      case 66: return SDM_TOTAL_ACTIVE_ENERGY;
+      case 67: return SDM_TOTAL_REACTIVE_ENERGY;
+      case 68: return SDM_L1_IMPORT_ACTIVE_ENERGY;
+      case 69: return SDM_L2_IMPORT_ACTIVE_ENERGY;
+      case 70: return SDM_L3_IMPORT_ACTIVE_ENERGY;
+      case 71: return SDM_L1_EXPORT_ACTIVE_ENERGY;
+      case 72: return SDM_L2_EXPORT_ACTIVE_ENERGY;
+      case 73: return SDM_L3_EXPORT_ACTIVE_ENERGY;
+      case 74: return SDM_L1_TOTAL_ACTIVE_ENERGY;
+      case 75: return SDM_L2_TOTAL_ACTIVE_ENERGY;
+      case 76: return SDM_L3_TOTAL_ACTIVE_ENERGY;
+      case 77: return SDM_L1_IMPORT_REACTIVE_ENERGY;
+      case 78: return SDM_L2_IMPORT_REACTIVE_ENERGY;
+      case 79: return SDM_L3_IMPORT_REACTIVE_ENERGY;
+      case 80: return SDM_L1_EXPORT_REACTIVE_ENERGY;
+      case 81: return SDM_L2_EXPORT_REACTIVE_ENERGY;
+      case 82: return SDM_L3_EXPORT_REACTIVE_ENERGY;
+      case 83: return SDM_L1_TOTAL_REACTIVE_ENERGY;
+      case 84: return SDM_L2_TOTAL_REACTIVE_ENERGY;
+      case 85: return SDM_L3_TOTAL_REACTIVE_ENERGY;
     }
```

Comment:
These switch statements will attribute to code increase. Not sure though what can be done to make this simpler.